configure.ac

```diff
@@ -5998,6 +6001,48 @@ then
     AM_CFLAGS="$AM_CFLAGS -DHAVE_DH_DEFAULT_PARAMS"
 fi
 
+
+AC_ARG_WITH([max-rsa-bits],
+    [AS_HELP_STRING([--with-max-rsa-bits=number],[number of bits to support for RSA, DH, and DSA keys])],
+    [WITH_MAX_ASSYM_KEY_BITS=$withval],
+    [WITH_MAX_ASSYM_KEY_BITS="$DEFAULT_MAX_ASSYM_KEY_BITS"])
+
+if test -n "$WITH_MAX_ASSYM_KEY_BITS"; then
+   if test "$WITH_MAX_ASSYM_KEY_BITS" -lt 1024 -o "$WITH_MAX_ASSYM_KEY_BITS" -gt 16384; then
+       AC_MSG_ERROR([--with-max-rsa-bits argument must be between 1024 and 16384 inclusive])
+   fi
+   AM_CFLAGS="$AM_CFLAGS -DRSA_MAX_SIZE=$WITH_MAX_ASSYM_KEY_BITS -DFP_MAX_BITS=$((WITH_MAX_ASSYM_KEY_BITS * 2)) -DSP_INT_BITS=$WITH_MAX_ASSYM_KEY_BITS"
+fi
+
+AC_ARG_WITH([max-ecc-bits],
+    [AS_HELP_STRING([--with-max-ecc-bits=number],[number of bits to support for ECC algorithms])],
+    [WITH_MAX_ECC_BITS=$withval],
+    [WITH_MAX_ECC_BITS="$DEFAULT_MAX_ECC_BITS"])
+
+if test -n "$WITH_MAX_ECC_BITS"; then
+   if test "$WITH_MAX_ECC_BITS" -lt 112 -o "$WITH_MAX_ECC_BITS" -gt 2048; then
+       AC_MSG_ERROR([--with-max-ecc-bits argument must be between 112 and 2048 inclusive])
+   fi
+   AM_CFLAGS="$AM_CFLAGS -DMAX_ECC_BITS=$WITH_MAX_ECC_BITS"
+fi
+
+
+AC_ARG_ENABLE([context-user-data],
```

Comment:
Is `context-user-data` the best name? Why not `--enable-user-data=` or `--enable-exdata=? Can the help explain that numbers can be provided?

Reply:
I named it `--enable-context-user-data` because `--enable-user-data` sounds like it could be something to do with protocols/transports stuff.  the name I picked is fugly but I think it'll save us from confused users.

I fixed the deficient help msg.  now reads
```Enables option for storing user-defined data in TLS API contexts, with optional argument the number of slots to allocate (default: disabled)```

Reply:
HAVE_EX_DATA means: have extra user data associated with the object.
Put extra in the name. ie. extra-user-data

Reply:
(done)